recipes/google-cloud-cpp/2.x/conanfile.py

```diff
@@ -138,23 +138,29 @@ def source(self):
         get(self, **self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True)
 
     def requirements(self):
-        self.requires("protobuf/3.21.9", transitive_headers=True)
-        self.requires("grpc/1.50.1", transitive_headers=True)
-        self.requires("nlohmann_json/3.10.0")
-        self.requires("crc32c/1.1.1")
-        self.requires("abseil/20220623.0", transitive_headers=True)
-        self.requires("libcurl/7.88.1")
+        # Updating Protobuf requires updates to the generated code, even within
+        # a minor version.
+        self.requires("protobuf/3.21.12", transitive_headers=True)
+        # Each release of Abseil is a new major version, not guaranteed to be
+        # compatible with previous verssions.
+        self.requires("abseil/[>=20230125.3.0 <20230126]", transitive_headers=True)
+        # The rest are more semver-like and require less pinning.
+        self.requires("grpc/[>=1.50]", transitive_headers=True)
+        self.requires("nlohmann_json/[>=3.10 <4]")
+        self.requires("crc32c/[>=1.1 <2]")
```

Comment:
We appreciate the comments for these ones, but besides libcurl, openssl and zlib, we're not _quite yet_ ready to allow general semver usage on libraries (But we're close to it!) - For now these should be pinned

Reply:
Ack.  Done.